src/pickup.h

```diff
@@ -2,6 +2,7 @@
 #define PICKUP_H
 
 #include "enums.h"
+#include "character.h"
```

Comment:
Why this? I don't see `Character` being used here. If you did this for the `player` parameter, a forward declaration of `class player;` is sufficient in the header.